### DISCUSSION TITLE
[아이디어] 어간 및 기본형 추출

### DISCUSSION THREAD

**Discussion by aajik** [bab2min/kiwipiepy#165] [신기능에 대한 아이디어]
예를 들면
text = "아버지 가방에 들어가신다."에서 '들어가신다'를 '들어가다'와 같이 기본형으로 추출하는 방법이 있으면 좋을 것같네요.
뭐, 아직 제가 kiwi에 대한 기능을 다 알지 못해서 이런 제안을 들릴 수는 있습니다.

**Comment by bab2min** [comment]
@aajik `들어가신다`를 `들어가/VV` + `시/EP` + `ㄴ다/EF`로 분석하는 대신 `들어가시/VV` + `ㄴ다/EF` 처럼 분석하는 옵션을 요청하시는거죠? 선어말어미는 분리하지 않고 어간에서 어말어미만 떼어내는식으로요~

**Reply by aajik** [reply]
네, 그렇습니다.
제가 사용하는 꼼수는 아래와 같습니다.

    for token in tokens :
        word = ""
        if( token.tag in ['NNG','NNP','NP','NR','NP',]) : # 체언(일반,고유,의존,수사,대명사)
            word = token.form
            pass
        elif( token.tag in ['VA','VA-I', 'VV', 'VV-R',]) : # 용어(형용사,동사)
            word = "{}다".format(token.form)
        elif( token.tag in ['VX',]) : # 용어(보조용언(VX))
            # word = "{}다".format(token.form)
            pass
        if( len(word) > 0 ) :
            clean_text += word + " "
            
    clean_text = clean_text.strip()

**Reply by bab2min** [reply]
아, 용언에 '-다'를 붙이는 기능이라면 `token.lemma`를 사용하시면 됩니다.
https://github.com/bab2min/kiwipiepy/blob/46961a8886669ff963bb5f0270e68aa18deaeb39/kiwipiepy/_c_api.pyi#L120-L124

**Reply by aajik** [reply]
헉, 저의 꼼수가 사고(오해)를 쳣네요.
용언에 '-다'를 붙이는 것이 아니고, 위 답변 주신 것이 맞습니다.
혼란을 드려 죄송...

들어가신다를 들어가/VV + 시/EP + ㄴ다/EF로 분석하는 대신 들어가시/VV + ㄴ다/EF 처럼 분석하는 옵션을 요청하시는거죠? 선어말어미는 분리하지 않고 어간에서 어말어미만 떼어내는식으로요

가 정확합니다.

**Reply by bab2min** [reply]
아 네 그렇군요 다음 업데이트 때 옵션으로 추가해보겠습니다~
